src/cmap/connection_pool.ts

```diff
@@ -300,35 +316,38 @@ class ConnectionPool extends EventEmitter {
     // drain the wait queue
     while (this.waitQueueSize) {
       const waitQueueMember = this[kWaitQueue].pop();
-      clearTimeout(waitQueueMember.timer);
-      if (!waitQueueMember[kCancelled]) {
-        waitQueueMember.callback(new MongoError('connection pool closed'));
+      if (waitQueueMember && waitQueueMember.timer) {
```

Comment:
you have the same bug you used to have below here, you're gating the entire logic here on the existence of the `timer`, but you only want to gate the `clearTimeout` call